src/components/user-defined-sets/sets.js

```diff
@@ -107,55 +114,40 @@ export function clearCurrentSet(state) {
 export function nameCurrentSet(state, name, clear) {
   return {
     ...state,
-    namedSets: state.namedSets.set(name, state.currentSet),
-    currentSet: (clear ? ImmutableSet() : state.currentSet),
+    namedSets: { ...state.namedSets, [name]: state.currentSet },
+    currentSet: (clear ? [] : state.currentSet),
   };
 }
 
 // TODO: better to do explicit checks than try-catch
-// TODO: serialization should go all the way to string
 // TODO: add tests when time to un-comment
 
-/* export function serialize(state) {
-  const {
-    namedSets = ImmutableOrderedMap(),
-    currentSet = ImmutableSet(),
-  } = state;
-  return {
-    namedSets: namedSets.toArray(),
-    currentSet: currentSet.toArray(),
-  };
-} */
-
-/* export function unserialize(serializedState) {
-  const {
-    namedSets = [],
-    currentSet = [],
-  } = serializedState;
-  return {
-    namedSets: ImmutableOrderedMap(namedSets),
-    currentSet: ImmutableSet(currentSet),
-  };
-} */
-
-/* export function persist(state, setTypeKey, datasetKey) {
-  const serializedState = serialize(state);
-  const serializedSets = JSON.parse(localStorage.getItem('sets')) || {};
-  try {
-    serializedSets[setTypeKey][datasetKey] = serializedState;
-  } catch (e) {
-    serializedSets[setTypeKey] = {
-      [datasetKey]: serializedState,
+/**
+ * @param {object} state A state object to persist to local storage.
+ * @param {string} setTypeKey The type of object in the state, for example "cells" or "genes".
+ * @param {string} datasetKey The dataset from which the sets come, for example "linnarson-2018".
+ */
+export function persist(state, setTypeKey, datasetKey) {
+  const sets = store.get('sets') || {};
+  const stateToSave = { namedSets: state.namedSets };
+  if (sets.hasOwnProperty(setTypeKey)) {
+    sets[setTypeKey][datasetKey] = stateToSave;
+  } else {
+    sets[setTypeKey] = {
+      [datasetKey]: stateToSave,
     };
   }
-  localStorage.setItem('sets', JSON.stringify(serializedSets));
-} */
+  store.set('sets', sets);
+}
 
-/* export function restore(setTypeKey, datasetKey) {
+/**
+ * @param {string} setTypeKey The type of object in the state, for example "cells" or "genes".
+ * @param {string} datasetKey The dataset from which the sets come, for example "linnarson-2018".
+ */
+export function restore(setTypeKey, datasetKey) {
   try {
-    const serializedState = JSON.parse(localStorage.getItem('sets'))[setTypeKey][datasetKey];
-    return unserialize(serializedState);
+    return { ...initialState, ...store.get('sets')[setTypeKey][datasetKey] };
   } catch (e) {
```

Comment:
If this is about the missing key, I'd prefer that that be checked explicitly, and leave try-catch for real error handling.